internal/controller/components/kueue/kueue_controller_actions.go

```diff
@@ -181,3 +181,39 @@ func manageKueueAdminRoleBinding(ctx context.Context, rr *odhtypes.Reconciliatio
 	log.Info("Successfully managed kueue admin role binding", "adminGroups", validAdminGroups)
 	return nil
 }
+
+func manageDefaultKueueResourcesAction(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	kueueCRInstance, ok := rr.Instance.(*componentApi.Kueue)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentApi.Kueue)", rr.Instance)
+	}
+
+	// Only proceed if Kueue is in Managed or Unmanaged state.
+	if kueueCRInstance.Spec.ManagementState != operatorv1.Managed && kueueCRInstance.Spec.ManagementState != operatorv1.Unmanaged {
+		return nil
+	}
+
+	// In Unmanaged case create the default ocp kueue configuration.
+	if kueueCRInstance.Spec.ManagementState == operatorv1.Unmanaged {
+		defaultKueueConfig := createKueueConfigurationCluster(kueueOperatorNamespace)
+		rr.Resources = append(rr.Resources, *defaultKueueConfig)
+	}
+
+	// Generate default ClusterQueue.
+	clusterQueue := createDefaultClusterQueue(kueueCRInstance.Spec.DefaultClusterQueueName)
+	rr.Resources = append(rr.Resources, *clusterQueue)
+
+	// Get all managed namespaces (i.e. the one opterd in with the addition of the proper labels).
+	managedNamespaces, err := getManagedNamespaces(ctx, rr.Client)
+	if err != nil {
+		return fmt.Errorf("failed to get managed namespaces: %w", err)
+	}
+
+	// Generate LocalQueues for managed namespaces.
+	for _, ns := range managedNamespaces {
```

Comment:
Since we're iterating over the namespaces, we should consider updating them to include both the old and new annotations. This is important because the opt-in logic may differ depending on the Kueue flavor (e.g., embedded or OpenShift).